Upgrade to Lucene 10.1.0

ChrisHegarty opened the pull request:
This commit upgrades to Lucene 10.1.0.

pmpailis committed:
Updating RankDocRetrieverBuilderIT for lucene_snapshot branch (#114098)

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-a4c0f741ccc

cbuescher committed:
Add Snowball upgrade changelog (#114146)

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-a4c0f741ccc

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-a4c0f741ccc

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-a4c0f741ccc

pmpailis committed:
muting RankDocsQueryBuilderTests testRankDocsQueryEarlyTerminate (#114198)

cbuescher committed:
Merge branch 'main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

javanna committed:
Delete full cluster restart test removed in main

This was restored due to a bad merge.

cbuescher committed:
Remove smoke test section in MetadataCreateIndexServiceTests

javanna committed:
spotless

javanna committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
Address EngineTestCase compile errors

javanna committed:
Address more compile errors around TotalHitCountCollectorManager constructor

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-a4c0f741ccc

javanna committed:
Merge branch 'main' into lucene_snapshot

cbuescher committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
Merge branch 'main' into lucene_snapshot

benwtrent committed:
Merge remote-tracking branch 'upstream/main' into lucene_snapshot

benwtrent committed:
Fix bbq for Lucene 10

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-a4c0f741ccc

cbuescher committed:
Merge branch 'main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-bc478d85a12

ChrisHegarty committed:
apply MADV_NORMAL advice to enable more aggressive readahead (#114410)

cbuescher committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-eadc07cc6a1

cbuescher committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
Remove empty queue conditional from slicing logic (#114513)

With recent changes in Lucene around not forking execution when not necessary, we have
removed the search worker thread pool in #111099. The worker thread pool had unlimited queue,
and the fear was that we couuld have much more queueing on the search thread pool if we use
it to parallelize execution across segments, because every shard would take up to a thread
per slice when executing the query phase.

We have then introduced an additional conditional to stop parallelizing when there is a queue.
That is perhaps a bit extreme, as it's a decision made when creating the searcher, while a
queue may no longer be there once the search is executing. This has caused some benchmarks
regressions, hence this commit removes the additional queue dependent conditional in order
to perform additional benchmarks without it.

cbuescher committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-eadc07cc6a1

cbuescher committed:
Merge branch 'main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

benwtrent committed:
Merge branch 'lucene_snapshot' of github.com:elastic/elasticsearch into lucene_snapshot

cbuescher committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-eadc07cc6a1

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-eadc07cc6a1

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-eadc07cc6a1

cbuescher committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-e821a1b4f87

cbuescher committed:
Merge branch 'main' into lucene_snapshot

cbuescher committed:
Put some duct tape on ES816BinaryFlatVectorsScorer compilation errors

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

benwtrent committed:
Merge branch 'lucene_snapshot' of github.com:elastic/elasticsearch into lucene_snapshot

ChrisHegarty committed:
Add new Index version for 10.0.1

benwtrent committed:
Merge branch 'lucene_snapshot' of github.com:elastic/elasticsearch into lucene_snapshot

benwtrent committed:
Correcting BinarizedByteVectorValues & OffHeapBinarizedVectorValues methods and overrides

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-5851033047d

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

javanna committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-5851033047d

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-5851033047d

cbuescher committed:
Merge branch 'main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-5851033047d

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-5851033047d

javanna committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-5851033047d

cbuescher committed:
Merge branch 'main' into lucene_snapshot

cbuescher committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-5851033047d

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-5851033047d

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-5851033047d

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-5851033047d

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-7a0365d4d39

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-7a0365d4d39

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-7a0365d4d39

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-7a0365d4d39

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-7a0365d4d39

cbuescher committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-7a0365d4d39

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

cbuescher committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-7a0365d4d39

cbuescher committed:
Fix wrong IndexVersion order of 10_0_1 update

cbuescher committed:
Merge branch 'main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

cbuescher committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-7a0365d4d39

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-7a0365d4d39

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-7a0365d4d39

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.1-snapshot-7a0365d4d39

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

javanna committed:
Move the lucene_snapshot branch to target branch_10x (#116242)

We have been tracking branch_10_0 for a bit following the Lucene 10.0 release.
With this commit we move to tracking 10x, so that we get ready to upgrade to
Lucene 10.1 when needed.

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-9a14d573c81

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

javanna committed:
Adjust compile errors due to CompoundFormat#getCompoundReader changes

Relates to https://github.com/apache/lucene/pull/13907

getCompoundReader no longer takes IOContext as its last argument.

javanna committed:
Adapt to removal of LeafSimScorer

Relates to https://github.com/apache/lucene/pull/13957

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-6e78f379d92

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

javanna committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
Adjust FieldUsageStatsIT following removal of LeafSimScorer

This is a leftover after adapting to the removal of LeafSimScorer

Relates to https://github.com/apache/lucene/pull/13957

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-6e78f379d92

javanna committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
Bump Elasticsearch codec to track Lucene101Codec (#116318)


See https://github.com/apache/lucene/pull/13968

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-6e78f379d92

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-6e78f379d92

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-6e78f379d92

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-ae92ce34c01

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-ffc94df1c2d

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-ffc94df1c2d

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-98918aa223e

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-a5d44d89eb1

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-a5d44d89eb1

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-a5d44d89eb1

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-c96ec0be678

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-a67120e1754

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-b70d214217a

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-816b5fa3c3b

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-cbec73e4c32

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-4f682ae6afc

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-4f682ae6afc

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot

ChrisHegarty committed:
lucene_snapshot: spotless

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

ChrisHegarty committed:
Remove deprecated source mode from DocumentParserContextTests

ChrisHegarty committed:
Update SyntheticSourceIndexSettingsProviderTests to expect source mode deprecation warning

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-71715b59e8a

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-71715b59e8a

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-c529c49c3e5

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-c529c49c3e5

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-b732543b575

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-b732543b575

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-b732543b575

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot

ChrisHegarty committed:
lucene_shapshot: Update DisiWrapper construction after apache/14023

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-8762de7f114

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-c9c631f9d29

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-895177861f3

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-895177861f3

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

javanna committed:
Fix completion postings format name

We moved to Completion101 with #116318, but that has been overwritten by

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-d2c2a44c583

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-d2c2a44c583

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-d2c2a44c583

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-3aefd605c43

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

javanna committed:
Fix binary quantized tests compile errors

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-4cdb67329fb

javanna committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
Adjust profile docs

javanna committed:
Adjust profile tests

javanna committed:
fix yaml compat test transform to address profile failure

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-559c4664eac

javanna committed:
Remove assertion that always fails on lucene_snapshot

javanna committed:
Address testRegexAcceleration failure

javanna committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-f8404620878

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-add9c09c84e

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.2.0-snapshot-d0866a699d6

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.2.0-snapshot-d0866a699d6

javanna committed:
Revert "[Automated] Update Lucene snapshot to 10.2.0-snapshot-d0866a699d6"

This reverts commit 986664b60040088136775d6cbd5e26d9893a5d87.

javanna committed:
Revert "[Automated] Update Lucene snapshot to 10.2.0-snapshot-d0866a699d6"

This reverts commit def22b7bea4ea8726217e04da40df08b94ef183f.

javanna committed:
Temporarily set lucene_snapshot build branch to branch_10_1

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-6a0531b0f48

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-6a0531b0f48

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-884954006de

javanna committed:
Merge branch 'main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-884954006de

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.0-snapshot-884954006de

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.1-snapshot-7415d5a7e87

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.1-snapshot-7415d5a7e87

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.1-snapshot-7415d5a7e87

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.1-snapshot-7415d5a7e87

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.1-snapshot-7415d5a7e87

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.1-snapshot-7415d5a7e87

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.1.1-snapshot-7415d5a7e87

ChrisHegarty committed:
Fix versions for Lucene 10.1.0

elasticsearchmachine commented:
Pinging @elastic/es-search (Team:Search)

elasticsearchmachine commented:
Hi @ChrisHegarty, I've created a changelog YAML for you.

ChrisHegarty committed:
Update docs/changelog/119308.yaml

ChrisHegarty committed:
Merge remote-tracking branch 'upstream/main' into lucene_snapshot_10_1

ChrisHegarty committed:
Update readCommit for N-2

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty commented on server/src/main/java/org/elasticsearch/common/lucene/Lucene.java:
```diff
@@ -156,25 +157,7 @@ public static SegmentInfos readSegmentInfos(IndexCommit commit) throws IOExcepti
-                e.addSuppressed(suppressed);
-            }
-            throw e;
-        }
+        return SegmentInfos.readCommit(directory, segmentsFileName, MIN_SUPPORTED_MAJOR);
```
@javanna @tlrx 

In preparation for merging Lucene 10.1, I updated the code here as per commit https://github.com/elastic/elasticsearch/pull/119308/commits/8d729160ce364b4919e19a467a371aaeecc276ef (it's a bit unfortunate that we didn't do this in the lucene_snapshot branch, but can do it now)

Given the relaxation to read N-2 in #118941, and the extensive comments and assertions here, then I think that this change is semantically correct. Can you please confirm, and whether or not an existing constant could be used rather than adding a new one here.

tlrx replied:
> Can you please confirm, and whether or not an existing constant could be used rather than adding a new one here.

Luca introduced `IndexVersions#MINIMUM_READONLY_COMPATIBLE`, I think it would be better to reuse it here. Otherwise the change LGTM, thanks Chris.

ChrisHegarty replied:
Thanks @tlrx - done in [8b0af43](https://github.com/elastic/elasticsearch/pull/119308/commits/8b0af43af26b64b8dbb9cc861a8fdb14d0be080b)

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Bump Elasticsearch codec to track Lucene101Codec

ChrisHegarty committed:
clean up

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
use IndexVersions.MINIMUM_READONLY_COMPATIBLE

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

javanna commented on server/src/main/java/org/elasticsearch/common/lucene/Lucene.java:
```diff
@@ -156,25 +154,7 @@ public static SegmentInfos readSegmentInfos(IndexCommit commit) throws IOExcepti
-            }
-            throw e;
-        }
+        return SegmentInfos.readCommit(directory, segmentsFileName, IndexVersions.MINIMUM_READONLY_COMPATIBLE.luceneVersion().major);
```
looks great, thanks!

javanna reviewed (approved):
LGTM thanks (only one small thing in the buildkite pipelines that I think should get reverted, although it should not take effect)

ChrisHegarty committed:
revert build-snapshot.yml

ChrisHegarty commented on .buildkite/pipelines/lucene-snapshot/build-snapshot.yml:
```diff
@@ -2,7 +2,9 @@ steps:
-    if: build.env("LUCENE_BUILD_ID") == null || build.env("LUCENE_BUILD_ID") == ""
+    if: (build.env("LUCENE_BUILD_ID") == null || build.env("LUCENE_BUILD_ID") == "")
+    build:
+      branch: branch_10_1
```
Reverted. [704e367](https://github.com/elastic/elasticsearch/pull/119308/commits/704e36737b4c769f90c18f34ec8c2bed604a98d5)

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

Commit:
[CI] Auto commit changes from spotless

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

mayya-sharipova committed:
Remove temp XIntervals (#119937)

PR #114234 introduced XIntervals as a temp solution while waiting for
a fix in Lucene 10.1. With Lucene 10.1 we don't need this class
anymore.

Related to #113554

chatman commented:
@ChrisHegarty I'm working on https://github.com/apache/lucene/pull/14131, but in order to test it with Elasticsearch, I needed to test with Elasticsearch that uses Lucene 10. Hence, I'm looking at this branch. However, the work in https://github.com/apache/lucene/pull/14131 integrates cuvs-java that uses Project Panama's features in JDK 22. However, Elasticsearch (as of this branch) still is on JDK21. Hence, trying to "gradlew compileJava" results in the following:
```ishan@4090-workstation ~/code/elasticsearch (lucene_snapshot_10_1) $ ./gradlew compileJava 
=======================================
Elasticsearch Build Hamster says Hello!
  Gradle Version        : 8.12
  OS Info               : Linux 6.8.0-49-generic (amd64)
  JDK Version           : 22.0.2+9-70 (Oracle)
  JAVA_HOME             : /opt/jdk-22.0.2
  Random Testing Seed   : 5173214438CDD58D
  In FIPS 140 mode      : false
=======================================

> Task :server:compileJava FAILED
warning: /home/ishan/.m2/repository/com/nvidia/cuvs/cuvs-java/25.02/cuvs-java-25.02.jar(/module-info.class): major version 66 is newer than 65, the highest major version supported by this compiler.
  It is recommended that the compiler be upgraded.
error: warnings found and -Werror specified
1 error
1 warning

[Incubating] Problems report is available at: file:///home/ishan/code/elasticsearch/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.
```
(the patch is https://termbin.com/v0co on top of this branch, along with the Lucene PR for cuVS being available from a local maven snapshot based on 10.2).

Any ideas or thoughts, please?

ChrisHegarty commented:
> Any ideas or thoughts, please? Is there a plan (or will there be) to raise minimum JDK level to 22?

There is certainly some complications here, but we can resolve them.

Elasticsearch compiles with a minimum of JDK 21, but runs with the latest JDK, currently 23. We mostly do something similar with Lucene. I need to look a bit more at how this is all supposed to work in Lucene first, then I'll see how we can get this working with ES.

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

chatman commented:
> Elasticsearch compiles with a minimum of JDK 21, but runs with the latest JDK, currently 23. We mostly do something similar with Lucene. I need to look a bit more at how this is all supposed to work in Lucene first, then I'll see how we can get this working with ES.

I'm trying to get a branch working with it: https://github.com/SearchScale/elasticsearch/tree/cuvs-integration
Mainly, upgraded things to minimum JDK22 (so that it doesn't complain that CuVS is compiled with JDK22). 
I'll send you some ideas on a dev setup on that CuVS-Java PR so that you can kindly test it out.

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty commented:
Linking this test PR that we need before merging the Lucene 10.1.0 upgrade - #121104

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10_1

tlrx commented:
Nice work @ChrisHegarty :clap:

mark-vieira commented:
> Any ideas or thoughts, please? Is there a plan (or will there be) to raise minimum JDK level to 22?

There are no plans to bump the minimum Java version at this time. The soonest we would do this would be with the release of Java 25, as our general policy is for minimum runtime version to be LTS versions.
